vars/gh.groovy

```diff
@@ -56,39 +52,23 @@ def call(Map args = [:]) {
     }
   }
 
-  if (ghLocation?.trim()) {
-    log(level: 'DEBUG', text: 'gh: get the ghLocation from cache.')
-  } else {
-    log(level: 'DEBUG', text: 'gh: set the ghLocation.')
-    ghLocation = pwd(tmp: true)
-  }
-
-  withEnv(["PATH+GH=${ghLocation}"]) {
-    if (forceInstallation || !isInstalled(tool: 'gh', flag: '--version', version: version)) {
-      if(isUnix()) {
-        downloadInstaller(ghLocation, version)
-      } else {
-        installTools([[ tool: 'gh', version: version, provider: 'choco']])
-      }
-    }
-    withCredentials([string(credentialsId: "${credentialsId}", variable: 'GITHUB_TOKEN')]) {
-      def flagsCommand = ''
-      if (flags) {
-        flags.each { k, v ->
-          log(level: 'DEBUG', text: "gh: k ${k} - v ${v}")
-          if (v instanceof java.util.ArrayList || v instanceof List) {
-            v.findAll { it }.each { value ->
-              flagsCommand += "--${k}='${normalise(value)}' "
-            }
-          } else {
-            if (v) {
-              flagsCommand += "--${k}='${normalise(v)}' "
-            }
+  withGhEnv(args) {
+    def flagsCommand = ''
+    if (flags) {
+      flags.each { k, v ->
+        log(level: 'DEBUG', text: "gh: k ${k} - v ${v}")
+        if (v instanceof java.util.ArrayList || v instanceof List) {
```

Comment:
java.util.ArrayList implements List

Reply:
I don't recall now why I added it, but we could give a try, bear in mind, this change is about moving the installation to a different step. So I almost added nothing but refactoring.